libs/features/mas/src/aem-fragment.js

```diff
@@ -273,6 +279,16 @@ export class AemFragment extends HTMLElement {
             { fields: {}, id, tags, settings },
         );
     }
+
+    async generatePreview() {
+        const { previewFragment } = await import('https://mas.adobe.com/studio/libs/fragment-client.js');
```

Comment:
Shouldn't we apply preview on the JSON of `aem-fragment`?
Assuming `aem-fragment` fetches already the right fragment for the correct locale, we could apply the preview logic without translation and return it in `get data`.
That way, the fragment cache would remain unchanged, but the consumer (e.g:, merch-card) would get the final content.

Reply:
if you move this to `get data` and use the data as input, you can safely deduce the surface from the fragment data.

Reply:
i don't get you i think we need to meet :-)